test/function_test.py

```diff
@@ -1484,6 +1484,46 @@ def f(a: int) -> list[str]: ...
     assert Function.from_name("app", "f")._get_schema() == expected_schema
 
 
+@pytest.mark.usefixtures("set_env_client")
+def test_function_supported_input_formats(client, servicer):
+    app = App("app")
+
+    @app.function(serialized=True)
+    def f(a): ...
+
+    @app.function(serialized=True, is_generator=True)
+    def g(a): ...
+
+    @app.function(serialized=True)
+    @modal.fastapi_endpoint()
+    def web_f(): ...
+
+    @app.cls(serialized=True)
+    class A:
+        @modal.method()
+        def f(self): ...
+
+        @modal.fastapi_endpoint()
```

Comment:
Nit: Just to test a slightly different code path compared to the `fastapi_endpoint` above.

```suggestion
        @modal.web_server(port=8081)
```